resources/lib/library.py

```diff
@@ -1316,52 +1532,91 @@ def sortDictionary( self, dictionary ):
     # === ADDON/SOURCE EXPLORER ===
     # =============================
     
-    def explorer( self, history, location, label, thumbnail, itemType ):
+    def explorer( self, history, location, label, thumbnail, itemType, isWidget = False ):
+        isLibrary = False
+        widgetType = None
+        addonType = None
+
         dialogLabel = label[0].replace( "  >", "" )
+        if len( label ) is not 1:
+            dialogLabel = label[0].replace( "  >", "" ) + " - " + label[ len( label ) - 1 ].replace( "  >", "" )
 
-        # Default action - create shortcut
         listings = []
         
         tree = DATA._get_overrides_skin()
+
+        # Shortcut to go 'up'
+        if len( label ) is 1:
```

Comment:
for comparing with numbers i would use == and != (even though -5 to 255 are singletons, but changing the number to something outside that range would break this function when using "is")